### PR TITLE
docs(migration-guide): added migration guide for Icon and IconButton

### PR DESCRIPTION
**Pull request by thebesson**
# Purpose of PR

Add migration guides for Icon and Icon button components.

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -261,9 +265,194 @@ import {
...
+
+Icon component in v4 has a simplified API. Simply import a required icon from `@contentful/f36-icons`.
+
+```tsx static=true
+import { ChevronLeftIcon } from '@contentful/f36-icons';
+
+<ChevronLeftIcon size="small" variant="primary" />;
+```
+
+In order to use an iicon from a third-party library, simply pass it as `as` prop:
```
Comment (burakukula) [inline MIGRATION.md:278]:
```suggestion
In order to use an icon from a third-party library, simply pass it as `as` prop:
```

File: MIGRATION.md [hunk 2]
```diff
@@ -261,9 +265,194 @@ import {
...
+```tsx static=true
+import { Icon } from '@contentful/f36-icons';
+import { MdAccessAlarm } from 'react-icons/md';
+
+const AccessAlarm = (props) => {
+  return <Icon {...props} as={MdAccessAlarm} variant="secondary" />;
+};
+```
+
+Or a custom icon can be rendered as a child with standart html tags:
```
Comment (burakukula) [inline MIGRATION.md:289]:
```suggestion
There is an option to render custom icon as a child with standard HTML tags:
```

**Review by massao — APPROVED** [review]
LGTM

**Review by burakukula — APPROVED** [review]

File: MIGRATION.md [hunk 3]
```diff
@@ -21,14 +25,14 @@ We separate only icons from the main package. If you need to use our icons, plea
 
 When using NPM
 
-```
+```bash
```
Comment (burakukula) [inline MIGRATION.md:28]: maybe we need to do `static=true` here?

**Review by bgutsol — APPROVED** [review]

File: MIGRATION.md (hunk 3 above)
Comment (massao) [reply MIGRATION.md:28]: Not really, this is only used on the github repo, not on the website

Comment (thebesson) [reply MIGRATION.md:28]: my spellchecker was complaining about code block without a type, I'll update it to sh

**Pull request merged**